modules/nextflow/src/main/groovy/nextflow/scm/RepositoryProvider.groovy

```diff
@@ -153,6 +153,12 @@ abstract class RepositoryProvider {
                     def message = config.auth ? "Check that the ${name.capitalize()} user name and password provided are correct" : "Provide your ${name.capitalize()} user name and password to access this repository"
                     throw new AbortOperationException("Forbidden -- $message")
                 }
+            case 404:
+                log.debug "Response status: $code -- ${connection.getErrorStream()?.text}"
+                String conString = connection.toString()
+                int indexNearUrlStart = conString.indexOf("//") - 10
+                int startIndex = conString.indexOf("http",indexNearUrlStart)
+                throw new AbortOperationException("Uri not found. This uri was attempted for api query: ${conString.substring(startIndex)}")
         }
```

Comment:
Hi, sorry for the late reply. Was wondering what's the rationale of this snippet. Could you please provide more details?

Reply:
Good catch! If you refer to this fizzy implementation, I replaced it with `connection.getURL()` in the recent commit.

Reply:
If you refer to adding support for 404 error, I don't consider it a must have, but it can be a nice option. Without this the message displayed to user is:
```
Cannot find `test_proj` -- Make sure exists a BitBucketServer repository at this address `https://bitbucket.server.com/scm/test_proj/test_repo`
```
This is misleading when dealing with unsupported repository provider, as we new this repository existed, so the information was of no use in such case.

The new message indicates clearer what the source of error is:
```
Uri not found. This uri was attempted for api query: https://bitbucket.biscrum.com/rest/api/1.0/projects/test_proj/repos/test_repo/raw/main.nf
```

Reply:
I'm ok with that, but it should go in separate PR since it modified a common behaviour.

Reply:
OK. I'll remove this from this branch and create a new PR.

Reply:
A separate pull request is created: https://github.com/nextflow-io/nextflow/pull/1341